www/src/components/homepage/homepage-blog-posts.js

```diff
@@ -114,28 +114,22 @@ const ViewAll = () => (
   </ViewAllStyle>
 )
 
-class HomepageBlogPosts extends Component {
-  desktopMediaQuery
+export default function HomepageBlogPosts({ posts }) {
+  const [desktopViewport, setDesktopViewport] = useState(false)
+  let desktopMediaQuery = null
 
-  state = {
-    desktopViewport: false,
-  }
-
-  componentDidMount = () => {
-    this.desktopMediaQuery = window.matchMedia(`(min-width: ${breakpoints[3]}`)
-    this.desktopMediaQuery.addListener(this.updateViewPortState)
-    this.setState({ desktopViewport: this.desktopMediaQuery.matches })
-  }
-
-  componentWillUnmount = () => {
-    this.desktopMediaQuery.removeListener(this.updateViewPortState)
-  }
+  useEffect(() => {
+    desktopMediaQuery = window.matchMedia(`(min-width: ${breakpoints[3]}`)
+    desktopMediaQuery.addListener(updateViewPortState)
+    setDesktopViewport(desktopMediaQuery.matches)
+    return () => desktopMediaQuery.removeListener(updateViewPortState)
+  }, [])
 
-  updateViewPortState = () => {
-    this.setState({ desktopViewport: this.desktopMediaQuery.matches })
+  const updateViewPortState = () => {
+    setDesktopViewport(desktopMediaQuery.matches)
   }
 
-  splitPostsToColumns = posts =>
+  const splitPostsToColumns = posts =>
```

Comment:
You can pull this out of the component since it doesn't have any references to props/state